packages/redux-routine/src/cast-error.js

```diff
@@ -7,8 +12,41 @@
  */
 export default function castError( error ) {
 	if ( ! ( error instanceof Error ) ) {
-		error = new Error( error );
+		error = new ReduxRoutineResponseError( error );
 	}
-
 	return error;
 }
+
+/**
+ * A custom error for handling non-string error responses.
+ *
+ * Error responses are exposed on the `response` property of the error instance.
+ *
+ * @param {*} errorResponse
+ * @param {...*} args
+ * @return {Error} An instance of ReduxRoutineResponseError
+ * @constructor
+ */
+export function ReduxRoutineResponseError( errorResponse, ...args ) {
```

Comment:
Is there any reason this could not be `class extends Error` ?

Reply:
Babel doesn't handle extending built-in classes.

Reply:
I can only speak from experience when I tried to extend built-ins in code I've written.  I've gotten compile errors or I've seen errors when throwing the exception.  Switched to doing ES5 and no issues.

Reply:
When I saw the [https://babeljs.io/docs/en/caveats#classes] babel docs, I figured that meant they don't fully support extending built-ins.

Reply:
I didn't see the caveats link edit 'til after. It does _sound_ like it'd be an issue, though the generated code when actually trying to extend `Error` looks much like what you've hand-recreated here.

Reply:
I'm not too familiar with the online babel generator, is it using the full babel browser compiler or does it simulate transpiling via a build tool? I wonder if that might be a factor here (i.e. the online tool you linked to considers the browser in use as well).  

It's also _possible_ that the issues I experienced have been resolved in the more recent version of Babel (and they just haven't updated their caveats yet).  Regardless, I'm not certain what harm there is in using ES5 in this case?

Reply:
Personally, I'd much prefer to just extend the class, but I'm wary based on my experience.

Reply:
Should note, your link is also referencing Babel version 6.25.  v7.2 has more boilerplate it looks like.